### PR TITLE
fix: KD-13974 consent listener returning undefined

### PR DESCRIPTION
**Pull request by Mavr1**

### CODE REVIEW

File: package/src/KetchServiceProvider/KetchServiceProvider.tsx [hunk 1]
```diff
@@ -205,8 +205,9 @@ export const KetchServiceProvider: React.FC<KetchServiceProviderParams> = ({
         break;
 
       case EventName.consent:
-        consent.current = JSON.parse(data.data || {}) as Consent;
-        parameters.onConsentUpdated?.(data.data);
+        const serializedData = JSON.parse(data.data || {}) as Consent;
```
Comment (webdevbyjoss) [inline package/src/KetchServiceProvider/KetchServiceProvider.tsx:208]: I'd renamed `serializedData` to something like `consent`, as it is actually de-serialized data

File: package/src/KetchServiceProvider/KetchServiceProvider.tsx [hunk 2]
```diff
@@ -205,8 +205,9 @@ export const KetchServiceProvider: React.FC<KetchServiceProviderParams> = ({
...
+        consent.current = serializedData;
+        parameters.onConsentUpdated?.(serializedData);
```
Comment (webdevbyjoss) [inline package/src/KetchServiceProvider/KetchServiceProvider.tsx:210]: we may even just do 
```javascript
parameters.onConsentUpdated?.(consent.current);
```

**Review by webdevbyjoss — APPROVED** [review]

**Review by jboileau99 — APPROVED** [review]
Approved pending the above comments are addressed

**Review by jboileau99 — APPROVED** [review]

**Comment by jboileau99** [comment]
@SocketSecurity ignore-all

**Review by jboileau99 — APPROVED** [review]

**Pull request merged**